logic/scorer/scorer.go

```diff
@@ -60,260 +34,247 @@ func points(lead model.Card, hand []model.Card, isCrib bool) int {
-			return onepair, 2 /* a pair*/
-		}
+	pSt, pPts := ps.accumulate()
+	res.totalPoints += pPts
+	res.scoreType |= pSt
+	res.valuesToCounts = ps.valuesToCounts
 
-		return doubleRunOfThree, 8 /* double run of 3 */
-	}
+	return res
+}
 
-	// check run of 4 (and middle set run of 3)
-	if (values[1]+1 == values[2]) && (values[2]+1 == values[3]) {
-		if (min+1 == values[1]) || (values[3]+1 == max) {
-			// there's no way all 5 can be, so we just check for 4
-			return run4, 4
-		}
-		// since we were checking for 4, but we found the middle three to be a run,
-		// then we can check the two ends for a double run
-		if min == values[1] || values[3] == max {
-			return doubleRunOfThree, 8 /* double run of 3 */
-		}
-		return run3, 3 /* run of 3 */
+func scoreFifteens(ptVals [numCardsToScore]int) (scoreType, int) {
+	if (ptVals[0]|ptVals[1]|ptVals[2]|ptVals[3]|ptVals[4])&1 == 0 {
+		// all even numbered cards => no fifteens possible
+		return 0, 0
 	}
 
-	numPairs := 0
-	for i := 1; i < len(values); i++ {
-		if values[i] == values[i-1] {
-			numPairs++
-		}
-	}
-	if numPairs == 2 {
-		return twopair, 4 /* two points per pair*/
+	sum := ptVals[0] + ptVals[1] + ptVals[2] + ptVals[3] + ptVals[4]
+	if sum == 15 {
+		// only one fifteen possible
+		return fifteen1, 2
+	} else if sum < 15 || sum > 46 {
+		return 0, 0
 	}
 
-	if values[3]-min == 2 {
-		// either two pair or a double run of three
-		// but we already checked two pair
-		return doubleRunOfThree, 8 /* double run of 3 */
-	} else if max-values[1] == 2 {
-		// either two pair or a double run of three
-		// but we already checked two pair
-		return doubleRunOfThree, 8 /* double run of 3 */
+	var numFifteens int
+
+	for i := 0; i < len(ptVals); i++ {
+		many := howManyAddUpTo(15-ptVals[i], ptVals[i+1:])
+		numFifteens += many
 	}
 
-	var st scoreType
-	pts := 0
+	if numFifteens > 0 {
+		return fifteen0 << numFifteens, numFifteens * 2
+	}
+	return 0, 0
+}
 
-	if ((min+1 == values[1]) && (values[1]+1 == values[2])) ||
-		((values[2]+1 == values[3]) && (values[3]+1 == max)) {
-		st = st | run3 /* run of 3 (and maybe a pair) */
-		pts += 3
+func howManyAddUpTo(goal int, ptVals []int) int {
+	if len(ptVals) == 0 {
+		return 0
 	}
 
-	if numPairs == 1 {
-		// we already checked for two above
-		st = st | onepair /* one pair */
-		pts += 2
+	var many int
+	for i, o := range ptVals {
+		if o == goal {
+			many++
+		} else if o < goal {
+			// o is less than the goal. See what we can find with it
+			subWith := howManyAddUpTo(goal-o, ptVals[i+1:])
+			many += subWith
+		}
 	}
 
-	return st, pts
+	return many
 }
 
-func scoreFlushesAndNobs(lead model.Card, hand []model.Card, isCrib bool) (scoreType, int) {
-	var st scoreType
-	pts := 0
-	handSuit := hand[0].Suit
-	isHandFlush := true
-	for _, c := range hand {
-		if c.Suit != handSuit {
-			isHandFlush = false
+func scoreRuns(values [numCardsToScore]int, valuesToCounts []uint8) (scoreType, int) {
+	for _, v := range values {
+		if valuesToCounts[v-1] > 0 {
```

Comment:
:mag: is this branch unnecessary?

Reply:
no; imagine we had a run of 4: 3, 2, 4, 5. If we don't check this, we'd find a run of 3 right away and return

Reply:
ah; the comment says `this is already part of a run; skip calculations`, but perhaps it would be more accurate to say `this card _will_ be part of a larger run, allow that to find the point values`.